R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+    hypothesis <- paste0(fit@ParTable$lhs[1], fit@ParTable$op[1], rhs, "= 0")
+  } else {
+    hypothesis <- encodeColNames(.bainCleanModelInput(options[["model"]]))   
+  }
+  
+  bainResult <- bain::bain(x = fit, hypothesis = hypothesis, fraction = fraction, standardize = standardized)
+  return(bainResult)
+}
+
+############################################
+### EXSTRACTING AND STORING RESULTS ########
+############################################
+
+.bainGetGeneralTestResults <- function(dataset, options, bainContainer, ready, type, variable = NULL, pair = NULL, testType = NULL){
+  
+  set.seed(options[["seed"]])
+  
+  if(type %in% c("onesampleTTest", "independentTTest")){
+    result <- .bainGetTTestResults(dataset, options, bainContainer, ready, type, variable, testType)
+  } else if(type == "pairedTTest"){
+    result <- .bainGetPairedTTestResults(dataset, options, bainContainer, ready, type, pair, testType) 
+  } else if (type %in% c("anova", "ancova", "regression", "sem")){
+    result <- .bainGetRegressionResults(dataset, options, bainContainer, ready, type)
+  }
+  
+  return(result) 
+}
+
+.bainGetTTestResults <- function(dataset, options, bainContainer, ready, type, variable, testType){
+  
+  if(!is.null(bainContainer[[variable]]))
+    return(bainContainer[[variable]]$object)
+  
+  variableData <- dataset[[variable]]
+  testValue <- format(options[["testValue"]], scientific = FALSE)  
+  
+  p <- try({
+    if(type == "onesampleTTest"){
+      bainResult <- .bainTTestRaw(options, x = variableData, nu = testValue, type = testType)
+    } else if(type == "independentTTest"){
+      levels <- base::levels(dataset[[options[["groupingVariable"]]]])
+      if (length(levels) != 2) {
+        g1 <- "1"
+        g2 <- "2"
+      } else {
+        g1 <- levels[1]
+        g2 <- levels[2]
+      } 
+      subDataSet <- dataset[, c(variable, options[["groupingVariable"]])]
+      group1 <- subDataSet[subDataSet[[options[["groupingVariable"]]]]== g1, variable]
+      group2 <- subDataSet[subDataSet[[options[["groupingVariable"]]]]== g2, variable]
+      bainResult <- .bainTTestRaw(options, x = group1, y = group2, type = testType)
+    }
+  })
+  
+  if (isTryError(p)) {
+    bainContainer$setError(gettextf("An error occurred in the analysis:<br>%1$s.", jaspBase::.extractErrorMessage(p)))
+    return()
+  }
+  
+  bainContainer[[variable]] <- createJaspState(bainResult, dependencies = c("testValue", "hypothesis"))
+  bainContainer[[variable]]$dependOn(optionContainsValue = list("variables" = variable))
+  return(bainContainer[[variable]]$object)
+}
+
+.bainGetPairedTTestResults <- function(dataset, options, bainContainer, ready, type, pair, testType){
+  
+  currentPair <- paste(pair, collapse = " - ")
+  
+  if(!is.null(bainContainer[[currentPair]]))
+    return(bainContainer[[currentPair]]$object)
+  
+  if (pair[[2]] != "" && pair[[1]] != pair[[2]] && pair[[1]] != "") {
```

Comment:
Will this give problems if one of the two is `""`?

Reply:
I don't think so, under that situation the output shows like this:
![image](https://user-images.githubusercontent.com/25059399/107944220-2020ed00-6f8e-11eb-8496-ccb56c3b35f2.png)